tokio/tests/uds_datagram.rs

```diff
@@ -88,8 +88,14 @@ async fn try_send_recv_never_block() -> io::Result<()> {
 
         match dgram1.try_send(payload) {
             Err(err) => match err.kind() {
-                io::ErrorKind::WouldBlock | io::ErrorKind::Other => break,
-                _ => unreachable!("unexpected error {:?}", err),
+                io::ErrorKind::WouldBlock => break,
+                _ => {
+                    let errno = err.raw_os_error().expect("Not an errno?");
+                    if errno == libc::ENOBUFS {
```

Comment:
It seems like the condition we really want is this:
```suggestion
                    if err.raw_os_error() == Some(libc::ENOBUFS) {
```
If it isn't an errno, then it's just some other unexpected error.

Reply:
That would work, if ENOBUFS is the errno returned by all OSes.  It's true on FreeBSD, but I didn't check anywhere else.  We may end up needing to whitelist a couple of different errnos.

Reply:
Well, I don't know what the full list is. Note that there are two instances of the error-checking, but you only changed one.

Reply:
Oops yes.  Changed both now.